### PR TITLE
InterceptLinks should iterate all links

### LINKED ISSUE #8: problem with IterateLinks()
Hi, I'm having a problem with IterateLinks.  Currently, it aborts the `for` loop when certain matching href are found .. it then doesn't iterate through the other links in the doc.  PR to come soon....

### PR DESCRIPTION
**Pull request by boyvinall**
- don't abort iteration certain href are matched

fixes go-humble/router#8

### CODE REVIEW

File: router.go [hunk 1]
```diff
@@ -203,16 +203,16 @@ func (r *Router) InterceptLinks() {
 		href := link.GetAttribute("href")
 		switch {
 		case href == "":
-			return
+
```
Comment (albrow) [inline router.go:211]: Can you please add a `continue` statement in each instance where you removed a `return` statement for the purpose of clarity?

**Comment by albrow** [comment]
@boyvinall thanks for this fix. I just left one comment but after it is addressed, I will merge.

File: router.go (hunk 1 above)
Comment (boyvinall) [reply router.go:211]: sure, ok

**Comment by albrow** [comment]
One last thing. The Karma tests were failing due to an unrelated issue which I fixed in https://github.com/go-humble/router/pull/11. Would you mind rebasing off master? Then I'll run the Karma tests using the branch for this PR and if they pass, merge it :)

**Comment by boyvinall** [comment]
try that @albrow 😄

**Comment by albrow** [comment]
Tests pass. Thanks again!

**Comment by boyvinall** [comment]
No problem, thanks to you! :grinning:

**Pull request merged**